includes/ProductCategory/Helper.php

```diff
@@ -258,12 +259,32 @@ public static function enqueue_and_localize_dokan_multistep_category() {
      *
      * @since 3.7.0
      *
-     * @param int $product_id
+     * @param int $product
      *
      * @return array
      */
-    public static function get_product_chosen_category( $product_id ) {
-        return get_post_meta( $product_id, 'chosen_product_cat', true );
+    public static function get_product_chosen_category( $product ) {
+        if ( ! $product instanceof WC_Product ) {
+            $product = wc_get_product( $product );
+        }
+
+        if ( ! $product ) {
+            return [];
+        }
+
+        $product_id = ( 'variation' === $product->get_type() || 'subscription_variation' === $product->get_type() ) ? $product->get_parent_id() : $product->get_id();
+
+        $chosen_product_cat = get_post_meta( $product_id, 'chosen_product_cat', true );
+
+        if ( ! is_array( $chosen_product_cat ) ) {
+            return $chosen_product_cat;
+        }
+
+        return array_filter(
+            $chosen_product_cat, function ( $item ) {
+				return ! empty( $item );
+			}
+        );
```

Comment:
Why are we not considering the category restriction?
- What are the usages/purposes?
- Why are business logic not centralised?